### PR TITLE
[new release] checkseum (0.2.0)

### PR DESCRIPTION
**Pull request by dinosaure**
Adler-32, CRC32 and CRC32-C implementation in C and OCaml

- Project page: <a href="https://github.com/mirage/checkseum">https://github.com/mirage/checkseum</a>
- Documentation: <a href="https://mirage.github.io/checkseum/">https://mirage.github.io/checkseum/</a>

##### CHANGES:

- fix cross-compilation with `dune -x windows` (mirage/checkseum#45, @dinosaure, @pirbo)
- add CRC-24 (mirage/checkseum#43, @dinosaure, @cfcs)
- factorize C stubs (as digestif)
- avoid clash of names when we use `checkseum.c`
  Any functions are prefixed by `checkseum_`
- fix META file (mirage/checkseum#39 & mirage/checkseum#41, @hannesm, @dinosaure)
  A test was added to see if runes (static C libraries) are available for
  MirageOS targets (freestanding & xen)
- provide a binary `checkseum` to _digest_ standard input or file
  `checkseum.checkseum` is available to compute check-sum of standard input
  or file. The tool is used only for debugging.
- clean distribution (mirage/checkseum#38, @dinosaure)
  `checkseum` depends only on `bigarray-compat`, `base-bytes` & `optint`
- `limits.h` is available on any targets (mirage/checkseum#37, @dinosaure, @pirbo)

### CODE REVIEW

**Comment by dinosaure** [comment]
#15971 should be merge before, then, I will add a constraint on `checkseum`.

File: packages/checkseum/checkseum.0.2.0/opam [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  [ "dune" "build" "-p" name "-j" jobs ]
+  [ "sh" "-exc" "echo \"xen_linkopts = \\\"-l:laolao/xen/liblaolao_xen_stubs.a\\\"\" >> _build/default/META.checkseum"]
+  [ "sh" "-exc" "echo \"freestanding_linkopts = \\\"-l:laolao/freestanding/liblaolao_freestanding_stubs.a\\\"\" >> _build/default/META.checkseum"]
+  [ "dune" "runtest" "-p" name "-j" jobs ] {with-test}
+]
+
+install: [
+  [ "dune" "install" "-p" name ]
+  [ "./test/test_runes.ml" ] {with-test}
+]
```
Comment (kit-ty-kate) [inline packages/checkseum/checkseum.0.2.0/opam]: I'm confused. Why is this here?

Comment (dinosaure) [reply packages/checkseum/checkseum.0.2.0/opam]: The test `test_runes` try to check if static C libraries for MirageOS are properly installed (so it can be run only after a `dune install`) and the `META` file is well-formed.

Comment (kit-ty-kate) [reply packages/checkseum/checkseum.0.2.0/opam]: Using `dune install` is highly not recommended as it requires unnecessary dependencies like `ocamlfind`. I'd suggest to remove this field entirely.

Comment (dinosaure) [reply packages/checkseum/checkseum.0.2.0/opam]: At least, we must have `ocamlfind` for `checkseum` due to the MirageOS compatibility. But as you like, it's just for testing - so I can keep it on my side.

**Comment by kit-ty-kate** [comment]
Thanks!
